code/contracts/ONEWallet.sol

```diff
@@ -7,14 +7,15 @@ contract ONEWallet {
     uint8 interval; // otp interval in seconds, default is 30
     uint32 t0; // starting time block (effectiveTime (in ms) / interval)
     uint32 lifespan;  // in number of block (e.g. 1 block per [interval] seconds)
-    uint8 maxOperationsPerInterval; // number of transactions permitted per OTP interval. Each transaction shall have a unique nonce. The nonce is auto-incremented within each interval
+    uint8 maxOperationsPerInterval; // number of transactions permitted per OTP interval. Each transaction shall have a unique nonce. The nonce is auto-incremented within each interval 
 
     // global mutable
     address payable lastResortAddress; // where money will be sent during a recovery process (or when the wallet is beyond its lifespan)
     uint256 dailyLimit; // uint128 is sufficient, but uint256 is more efficient since EVM works with 32-byte words.
     uint256 spentToday; // note: instead of tracking the money spent for the last 24h, we are simply tracking money spent per 24h block based on UTC time. It is good enough for now, but we may want to change this later.
-    uint32 lastTransferDay;
+    uint32 lastTransferDay; // I'd use uint instead
```

Comment:
uint32 is enough to represent days. For gas efficiency, I noticed a trend in newer versions of solidity that the gas discrepancy between 256-bit variables and smaller ones are decreasing over time. Some of these can also be packed into a struct to further optimize gas. https://ethereum.stackexchange.com/questions/3067/why-does-uint8-cost-more-gas-than-uint256?noredirect=1&lq=1

Reply:
ok, stay with uint8, just be sure that it will not overflow.

Reply:
Marking this as resolved